stannp/stannp.go

```diff
@@ -142,12 +144,60 @@ func (s *Stannp) post(ctx context.Context, inputReader io.Reader, inputURL strin
 
 	res, err := s.client.Do(req)
 	if err != nil {
-		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req), false)
+		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req))
 	}
 
 	return res, nil
 }
 
+func (s *Stannp) GetPDFContents(ctx context.Context, pdfURL string) (*letter.PDFRes, *util.APIError) {
+	if !strings.HasPrefix(pdfURL, PDFURLPrefix) {
+		return nil, util.BuildError(400, fmt.Sprintf("pdfURL must begin with [%s]. your input was [%s]", PDFURLPrefix, pdfURL))
+	}
+
+	fileURL, err := url.Parse(pdfURL)
+	if err != nil {
+		return nil, util.BuildError(500, err.Error())
+	}
+	path := fileURL.Path
+	urlSegments := strings.Split(path, "/")
+	fileName := urlSegments[len(urlSegments)-1]
+
+	pdfGetReq, reqErr := http.NewRequestWithContext(ctx, http.MethodGet, pdfURL, nil)
+	if reqErr != nil {
+		return nil, util.BuildError(500, reqErr.Error())
+	}
+
+	resp, err := s.client.Do(pdfGetReq)
+	if err != nil {
+		return nil, util.BuildError(500, err.Error())
+	}
+
+	return &letter.PDFRes{
+		Contents: resp.Body,
+		Name:     fileName,
+	}, nil
+}
+
+func (s *Stannp) SavePDFContents(pdfContents io.ReadCloser) (*os.File, *util.APIError) {
```

Comment:
This doesn't need to be an `io.ReadCloser`. An `io.Reader` would suffice. The caller should handle closing as it sees fit.
More importantly, I question whether this method is a concern for this client.

Reply:
it's definitely not needed immediately for us but figured it would be decent for other users. will merge for now and re-evaluate.